modnet/models.py

```diff
@@ -554,79 +566,904 @@ class OR only return the most probable class.
+            with ZipFile(filename, "r") as zf:
+                namelist = zf.namelist()
+                _files = [_ for _ in namelist if not _.startswith("__MACOSX/") or _.startswith(".DS_STORE")]
+                if len(_files) == 1:
+                    with zf.open(_files.pop()) as f:
+                        pickled_data = pd.read_pickle(f)
+
+        if pickled_data is None:
+            pickled_data = pd.read_pickle(filename)
+
+
+        if isinstance(pickled_data, MODNetModel):
+            if not hasattr(pickled_data, "__modnet_version__"):
+                pickled_data.__modnet_version__ = "unknown"
+            pickled_data._restore_model()
+            LOG.info(f"Loaded {pickled_data} object, created with modnet version {pickled_data.__modnet_version__}")
+            return pickled_data
+
+        raise ValueError(
+            f"File {filename} did not contain compatible data to create a MODNetModel object, "
+            f"instead found {pickled_data.__class__.__name__}."
+        )
+
+def validate_model(train_data = None,
+                   val_data = None,
+                   targets = None,
+                   weights = None,
+                   num_classes=None,
+                   n_feat = 100,
+                   num_neurons = [[8],[8],[8],[8]],
+                   lr=0.1,
+                   batch_size = 64,
+                   epochs = 100,
+                   loss='mse',
+                   act = 'relu',
+                   out_act= 'linear',
+                   xscale='minmax',
+                   callbacks = [],
+                   preset_id = None,
+                   fold_id = None,
+                   verbose = 0,
+            ):
+
+    #deprecated
+    # data type can get messed up when passed to new process
+    #train_data.df_featurized = train_data.df_featurized.apply(pd.to_numeric)
+    #train_data.df_targets = train_data.df_targets.apply(pd.to_numeric)
+    #if val_data is not None:
+    #    val_data.df_featurized = val_data.df_featurized.apply(pd.to_numeric)
+    #    val_data.df_targets = val_data.df_targets.apply(pd.to_numeric)
+    #verbose=1
+
+    model = MODNetModel(
+        targets,
+        weights,
+        num_neurons=num_neurons,
+        n_feat=n_feat,
+        act=act,
+        out_act=out_act,
+        num_classes=num_classes
+    )
+
+    model.fit(
+        train_data,
+        lr = lr,
+        epochs = epochs,
+        batch_size = batch_size,
+        loss = loss,
+        xscale=xscale,
+        callbacks = callbacks,
+        verbose = verbose,
+        val_fraction = 0,
+        val_data = val_data,
+    )
+
+
+    learning_curve = model.history["val_loss"]
+
+    val_loss = model.evaluate(val_data)
+
+    #save model
+    model._make_picklable()
+
+    return val_loss, learning_curve, model, preset_id, fold_id
+
+
+def map_validate_model(kwargs):
+    return validate_model(**kwargs)
+
+
+#### Probabilistoc models####
+
+class Bayesian_MODNetModel(MODNetModel):
```

Comment:
We should stick to PEP naming conventions:
```suggestion
class BayesianMODNetModel(MODNetModel):
```